src/server/scripts/Northrend/Ulduar/HallsOfStone/instance_halls_of_stone.cpp

```diff
@@ -173,6 +184,15 @@ class instance_halls_of_stone : public InstanceMapScript
             if (type < MAX_ENCOUNTER)
                 Encounter[type] = data;
 
+            if (type == BOSS_MAIDEN_OF_GRIEF && data == DONE)
+                isMaidenOfGriefDead = true;
+            if (type == BOSS_KRYSTALLUS && data == DONE)
+                isKrystalusDead = true;
```

Comment:
You could improve this statement in this way:
```C++
if (data == DONE) {
  isMaidenOfGriefDead = (type == BOSS_MAIDEN_OF_GRIEF);
  isKrystalusDead     = (type == BOSS_KRYSTALLUS);
}
```

This change avoids to check the condition `(data == DONE)` twice and make the code more compact.
It's up to you to make this change or not, it's just a little tip.

Reply:
That is so cool! Never thought about it, thanks! I'll update it tomorrow.

Reply:
I tested this check and the door stays open, doesn't close 🤔